### PR TITLE
Tests refactoring and small improvements

### PR DESCRIPTION
**Pull request by notanengineercom**
I tried to group all the tests that were randomly spread on all the issues folder into specs. The tests suite is not finished, but this is like a part one of the series. I thought it made sense to order a bit the tests before implementing new features - like the `.configure()` one. On the process of doing so, I found the issue #97.

---
- Removed the node types - they were not in use
- Upgraded dependencies
- Removed unnecessary / old configs

In following prs I will upgrade to the latest typescript, so we can use optional chaining and nullish coalescing. I won't replace any types on the Transformation type file, to avoid dropping support on users that are using older versions of typescript.

### CODE REVIEW

File: src/Arguments.ts [hunk 1]
```diff
@@ -37,18 +37,15 @@ export class Arg {
...
+    static any<T extends 'string' | 'number' | 'boolean' | 'symbol' | 'undefined' | 'object' | 'function' | 'array'>(type: T): Argument<any> & any
     static any(type?: string): Argument<any> & any {
         const description = !type ? '{any arg}' : '{type ' + type + '}';
         return new Argument<any>(description, x => {
-            if(!type)
+            if (!type)
                 return true;
 
-            if(typeof x === 'undefined')
-                return true;
```
Comment (notanengineercom) [inline src/Arguments.ts:48]: I removed this as I see it as a bug. If we explicitly pass in the type on `Arg.any(type: T)`, we should always compare the type of x to it. Also if we expect x to be undefined both `Arg.any()` and `Arg.any('undefined')` will be true.

Comment (notanengineercom) [reply src/Arguments.ts:48]: But maybe it's there for a reason I am overseeing. So that's why I want to ask if it's okay to remove.

Comment (ffMathy) [reply src/Arguments.ts:48]: Perfect! Well, as long as there is no test describing why the change is there, I'd say we should remove it :+1:

Nicely done.

**Comment by ffMathy** [comment]
I really like where this is going, but it would be nice if every test would have a prefix with the issue number, so that we can know when the test was introduced.

Perhaps this can be inferred from a Blame - what do you think?

The reason I like it, is that it might not be clear why a certain test exists. Especially in a project like this.

I may be exaggerating the importance of this though :sweat_smile:

File: src/Arguments.ts (hunk 1 above)
Comment (notanengineercom) [reply src/Arguments.ts:48]: Cool! I will also add an Arguments spec, so we can have it tested 😄

**Comment by notanengineercom** [comment]
Thanks 😄 
> it would be nice if every test would have a prefix with the issue number, so that we can know when the test was introduced

Sure, we could do that. But how about instead of adding it to the test description, adding it as comment?
Example:

```ts
test('......', t => {
  // Issue 123: https://github.issue.url
})
```

This way the tests will be grouped in specs and the metadata of a linked issue if aplicable

**Comment by ffMathy** [comment]
That's even better - I love that.

Sorry if this seems like a big change to make. But I fear that if we don't have the direct "link" to every issue, we may think in the future "Why is this test here? It breaks something now and doesn't seem to add value." - and then we might correct it thinking that the test was wrong, or something similar.

I hope you understand :+1:

**Comment by notanengineercom** [comment]
Will do that then! And I think it won't be such a big change, so no worries on that 😄. And I agree with what you said, the better we document and add inform the tests, the safer it will be. Thanks for the feedback 👍

**Comment by notanengineercom** [comment]
Oh and the fix for the linked issue will happen on another pr. I think this pr is big enough already!

**Pull request merged**